src/components/Confirm/__tests__/Confirm.test.tsx

```diff
@@ -0,0 +1,232 @@
+  },
+  nextStep: jest.fn(),
+}
+let props = defaultProps
+
+const onfidoSuccessResponse = Promise.resolve({ onfidoSuccessResponse: {} })
+const onfidoErrorResponse = Promise.reject({
+  status: 422,
+  response: JSON.stringify({
+    error: {
+      message: 'There was a validation error on this request',
+      type: 'validation_error',
+      fields: { detect_glare: ['glare found in image'] },
+    },
+  }),
+})
+const continueWithOnfidoSubmission = Promise.resolve({
+  continueWithOnfidoSubmission: true,
+})
+
+const UPLOAD_TYPES = [
+  {
+    type: 'document',
+    method: 'document',
+    variant: '',
+    uploadFunction: uploadDocument,
+  },
+  {
+    type: 'selfie',
+    method: 'face',
+    variant: 'standard',
+    uploadFunction: sendMultiframeSelfie,
+  },
+  {
+    type: 'video',
+    method: 'face',
+    variant: 'video',
+    uploadFunction: uploadLiveVideo,
+  },
+]
+
+describe('Confirm', () => {
+  describe('onSubmitCallback', () => {
+    UPLOAD_TYPES.forEach(({ type, method, variant, uploadFunction }) => {
+      describe(`for ${type}`, () => {
+        afterEach(() => {
+          jest.clearAllMocks()
+          props = defaultProps
+        })
+
+        describe('when response contains onfidoSuccessResponse', () => {
+          beforeEach(() => {
+            props = {
+              ...defaultProps,
+              method,
+              capture: { ...defaultProps.capture, variant },
+              enterpriseFeatures: {
+                onSubmitDocument: () => onfidoSuccessResponse,
+                onSubmitSelfie: () => onfidoSuccessResponse,
+                onSubmitVideo: () => onfidoSuccessResponse,
+              },
+            }
+          })
+
+          it('does not trigger the SDK to send the request', async () => {
+            const spyUpload = jest.spyOn(
+              { [uploadFunction.name]: uploadFunction },
+              uploadFunction.name
+            )
+            const wrapper = mount(
+              <MockedReduxProvider>
+                <MockedLocalised>
+                  <Confirm {...props} />
+                </MockedLocalised>
+              </MockedReduxProvider>
+            )
+            await wrapper.find('.button-primary').simulate('click')
+
+            expect(spyUpload).not.toHaveBeenCalled()
+          })
+
+          it('triggers the onApiSuccess method', async () => {
+            // Spying on prop setCaptureMetadata called within onApiSuccess because we
+            // can't directly spy on onApiSuccess when using arrow function for class property
+            const spyOnApiSuccess = jest.spyOn(
+              props.actions,
+              'setCaptureMetadata'
+            )
+            const wrapper = mount(
+              <MockedReduxProvider>
+                <MockedLocalised>
+                  <Confirm {...props} />,
+                </MockedLocalised>
+              </MockedReduxProvider>
+            )
+            await wrapper.find('.button-primary').simulate('click')
```

Comment:
I don't know why but why `await` shouldn't be here (as `tsserver` complains `'await' has no effect on the type of this expression.`) but it doesn't work without `await`.

Normally I tend to use `wrapper.update` and `runAllPromises()` with:

```
const runAllPromises = new Promise(setImmediate)
```

as [pointed here](https://stackoverflow.com/a/51045733), but apparently it doesn't work here. Can you please figure out why?

Reply:
Not sure why it wasn't working for you but I got it working as described in that link. The only problem I had was an error being throw when the `onfidoErrorResponse` was defined so I've removed that part and just used `Promise.reject` directly in the error callbacks

Reply:
@eraxyz yeah I meant `runAllPromises` worked but the test still failed. I forgot that I have to handle the rejects instead of resolves as normal.